### PR TITLE
Add support for the Datetime type for MySQL

### PR DESCRIPTION
**Pull request by Eijebong**
This is mostly the same as for the Timestamp type so I reused both FromSql and
ToSql implementations.

### CODE REVIEW

File: diesel/src/mysql/types/date_and_time.rs [hunk 1]
```diff
@@ -41,10 +41,24 @@ macro_rules! mysql_time_impls {
     }
 }
 
+mysql_time_impls!(Datetime);
+primitive_impls!(Datetime -> NaiveDateTime);
 mysql_time_impls!(Timestamp);
 mysql_time_impls!(Time);
 mysql_time_impls!(Date);
 
+impl ToSql<Datetime, Mysql> for NaiveDateTime {
```
Comment (sgrif) [inline diesel/src/mysql/types/date_and_time.rs]: Could this be `impl<T: ToSql<Timestamp, Mysql>> ToSql<Datetime, Mysql> for T`?

Comment (Eijebong) [inline diesel/src/mysql/types/date_and_time.rs]:
```
error[E0119]: conflicting implementations of trait `types::FromSql<mysql::types::Datetime, mysql::backend::Mysql>` for type `std::borrow::Cow<'_, _>`:
  --> src/mysql/types/date_and_time.rs:56:1
   |
56 | / impl<T: ToSql<Timestamp, Mysql>> FromSql<Datetime, Mysql> for T {
57 | |     fn from_sql(bytes: Option<&[u8]>) -> Result<Self, Box<Error+Send+Sync>> {
58 | |         <T as FromSql<Timestamp, Mysql>>::from_sql(bytes)
59 | |     }
60 | | }
   | |_^ conflicting implementation for `std::borrow::Cow<'_, _>`
   | 
  ::: src/types/impls/primitives.rs
   |
91 | / impl<'a, T: ?Sized, ST, DB> FromSql<ST, DB> for Cow<'a, T> where
92 | |     T: 'a + ToOwned,
93 | |     DB: Backend + HasSqlType<ST>,
94 | |     T::Owned: FromSql<ST, DB>,
...  |
98 | |     }
99 | | }
   | |_- first implementation here
```

Comment (sgrif) [inline diesel/src/mysql/types/date_and_time.rs]: :+1:

**Comment by alanpoon** [comment]
The trait `diesel::Expression` is not implemented for `diesel::types::Datetime`, when I try to derive Insertable a Row struct that has field of type diesel::mysql::types::Datetime.

**Pull request merged**